### PR TITLE
Add default_scale to MeanScaler and enable the option in DeepAR-PyTorch

### PR DESCRIPTION
**Pull request by shubhamkapoor**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/torch/modules/scaler.py [hunk 1]
```diff
@@ -38,12 +38,20 @@ class MeanScaler(nn.Module):
 
     @validated()
     def __init__(
-        self, dim: int, keepdim: bool = False, minimum_scale: float = 1e-10
+        self,
+        dim: int,
+        keepdim: bool = False,
+        default_scale: Optional[float] = None,
```
Comment (lostella) [inline src/gluonts/torch/modules/scaler.py:45]: Maybe this could be added to the docstring above?

Comment (shubhamkapoor) [reply src/gluonts/torch/modules/scaler.py:45]: Done! I think the docstring of minimum_scale was wrong. Updated it as well. Please take a look.

File: src/gluonts/torch/model/deepar/estimator.py [hunk 2]
```diff
@@ -126,6 +126,10 @@ class DeepAREstimator(PyTorchLightningEstimator):
         (default: ``NegativeLogLikelihood()``).
     scaling
         Whether to automatically scale the target values (default: true).
+    default_scale
+        Default scale that is applied if the context length window is
+        completely unobserved. If not set, the scale in this case will be
+        the mean scale in the batch.
```
Comment (lostella) [inline src/gluonts/torch/model/deepar/estimator.py:132]: I think we probably want to fix this: having the model behaviour on one batch element depend on what the rest of the batch looks like is probably not desirable, right?

File: src/gluonts/torch/modules/scaler.py [hunk 3]
```diff
@@ -31,19 +31,29 @@ class MeanScaler(nn.Module):
     keepdim
         controls whether to retain dimension ``dim`` (of length 1) in the
         scale tensor, or suppress it.
-    minimum_scale
+    default_scale
         default scale that is used for elements that are constantly zero
+    minimum_scale
+        minimum possible scale that is used for any item.
         along dimension ``dim``.
```
Comment (lostella) [inline src/gluonts/torch/modules/scaler.py]: Forgot to remove a line?

File: src/gluonts/torch/modules/scaler.py [hunk 4]
```diff
@@ -57,7 +67,10 @@ def forward(
         denominator = torch.max(
             total_observed, torch.ones_like(total_observed)
         )
-        default_scale = weighted_sum.sum(dim=0) / denominator
```
Comment (lostella) [inline src/gluonts/torch/modules/scaler.py:60]: I think this was wrong in the first place, mixing things across dimension 0 should not occur probably

**Review by lostella — COMMENTED** [review]
I'm wondering whether we should just remove the previous behaviour to determine the `default_scale`, and default the new `default_scale` to `1.0` (too many "default" here, I know)

File: src/gluonts/torch/modules/scaler.py (hunk 3 above)
Comment (shubhamkapoor) [reply src/gluonts/torch/modules/scaler.py]: Done!

File: src/gluonts/torch/model/deepar/estimator.py (hunk 2 above)
Comment (shubhamkapoor) [reply src/gluonts/torch/model/deepar/estimator.py:132]: I think the reason for using batch statistics for default_scale makes sense from the point of the computation of the average batch loss. A single time series on a very high scale, if not scaled with appropriate a batch statistic, can mess up the loss, no?

**Review by lostella — APPROVED** [review]

**Review by lostella — APPROVED** [review]
Thanks! Adding the default_scale option makes total sense.

If we want to challenge the “batch-wise” default_scale computation that we already had, we can do that separately
